### PR TITLE
can.Component now supports lexical semantics for <content>

### LINKED ISSUE #1069: Lexical semantics for can.Component
`can.Component` should have lexical semantics for its light dom. Consider this fiddle: http://jsfiddle.net/5D4FK/2/

Lexical semantics would shield the user from internal implementation details of the component. Likewise, they would shield the Component itself from unintended (or even intended) injection from the outside. Conceptually, the user should never know/care about the internal details of a component's shadow dom and interact with it purely through its exposed API (attributes and the light DOM, rendered in the lexical template context).

### PR DESCRIPTION
**Pull request by zkat**
Fixes #1069

### CODE REVIEW

**Comment by daffl** [comment]
This is backwards compatible right? Can you rebase and double check why the tests are failing?

**Comment by zkat** [comment]
@daffl this is up to date and clear.

File: component/component.md [hunk 1]
```diff
@@ -263,6 +263,50 @@ only renders friendly messages:
       }
     });
 
+### LeakScope
```
Comment (justinbmeyer) [inline component/component.md]: I think leakScope should have it's own page, similar to how all the other prototype properties do.

Comment (justinbmeyer) [inline component/component.md]: Perhaps you can just move this into it's own MD file.

**Pull request merged**